src/query_ctrl.ts

```diff
@@ -464,10 +590,28 @@ export class InstanaQueryCtrl extends QueryCtrl {
     }
   }
 
+  adjustServiceEndpointSelectionPlaceholder() {
+    this.serviceEndpointSelectionText = this.uniqueEntities.length > 0
+      ? 'Please select (' + this.target.availableServicesEndpoints.length + '/' + this.uniqueEntities.length + ')'
+      : this.EMPTY_DROPDOWN_TEXT;
+  }
+
+  onNamefilterChanges() {
+    if (!this.target.serviceNamefilter) {
+      this.target.availableServicesEndpoints = this.uniqueEntities;
+    } else {
+      this.target.availableServicesEndpoints =
+        _.filter(this.uniqueEntities, entity => entity.label.includes(this.target.serviceNamefilter));
+    }
+
+    this.adjustServiceEndpointSelectionPlaceholder();
+    this.panelCtrl.refresh();
+  }
+
   onGroupChange() {
-    if (this.target.group && this.isApplication()) {
+    if (this.target.group && this.isAnalyzeApplication()) {
       this.target.showGroupBySecondLevel = this.target.group.key === 'call.http.header';
```

Comment:
Why only one field as secondary option ?
It should be all fields with "type":"KEY_VALUE_PAIR" from server.

Wishlist would be at least for the most commonly used ones:
- docker.label
- kubernetes.pod.label
- call.http.params
- kubernetes.label
- aws.ec2.tag

Reply:
Hi asafarian, good hint. We will check this for release 2.4.2